beam/www/beam/components/ControlButtons.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+	<div class="control-buttons">
+		<button @click="onCreate" :disabled="docstatus !== -1">Save</button>
+		<button @click="onSubmit" :disabled="docstatus !== 0">Submit</button>
+		<button @click="onCancel" :disabled="docstatus !== 1">Cancel</button>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { defineProps, ref } from 'vue'
+const docstatus = ref(-1)
+
+const props = defineProps<{
+	onCreate: () => Promise<{ data: any }>
```

Comment:
I think this button will involve both create and update actions. Maybe rename this to `onSave` instead?